apps/electron/layers/preload/src/index.ts

```diff
@@ -30,6 +31,30 @@ contextBridge.exposeInMainWorld('apis', {
 
   onSidebarVisibilityChange: (visible: boolean) =>
     ipcRenderer.invoke('ui:sidebar-visibility-change', visible),
+  signIn: () => {
+    electronGoogleOauth.openAuthWindowAndGetTokens();
+  },
+  ipcRenderer: {
+    on: (channel: string, listener: (event: any, ...args: any[]) => void) => {
+      const validChannels = ['send-token'];
+      if (validChannels.includes(channel)) {
+        // Deliberately strip event as it includes `sender` and is a security risk
+        // @ts-ignore
+        ipcRenderer.on(channel, (event, ...args) => listener(...args));
+      }
+    },
+    off: (channel: string) => {
+      ipcRenderer.removeAllListeners(channel);
+    },
+  },
+  reload: () => {
+    // const remote.BrowserWindow.getAllWindows();
+    const mainWindow = remote.BrowserWindow.getAllWindows().find(
```

Comment:
I think we can use existing reloading logic in the render after login. We will only have one window object at a time.